shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java

```diff
@@ -35,9 +38,19 @@
 public class FlutterView extends FrameLayout {
   private static final String TAG = "FlutterView";
 
+  // View configuration
   @NonNull
   private RenderMode renderMode;
 
+  // View structure
+  @Nullable
+  private FlutterRenderer.RenderSurface renderSurface;
+
+  // Flutter engine
+  @Nullable
+  private FlutterEngine flutterEngine;
+  private boolean isAttachedToFlutterEngine = false;
```

Comment:
Yes. I thought that an explicit variable would be more clear than a null check, because with a null check it isn't clear if that means attachment, or if there are other reasons why it might be null.

Reply:
I'd recommend converting `isAttachedToFlutterEngine` into a function that checks `flutterEngine` in order to reduce redundancy

Reply:
Done